src/cpp/cpp_scope.cpp

```diff
@@ -144,44 +140,48 @@ void cpp_scopet::lookup_rec(
 
     // Recursive call.
     // Note the different kind!
-    other_scope.lookup_rec(base_name, QUALIFIED, id_class, id_set);
+    other_scope.lookup_rec(
+      base_name_to_lookup, QUALIFIED, identifier_class, id_set);
   }
 
   if(kind==QUALIFIED)
     return; // done
 
-  if(!id_set.empty() &&
-     id_class!=id_classt::TEMPLATE) return; // done, upwards scopes are hidden
+  if(!id_set.empty() && identifier_class != id_classt::TEMPLATE)
+    return; // done, upwards scopes are hidden
 
   // ask parent, recursive call
   if(!is_root_scope())
-    get_parent().lookup_rec(base_name, kind, id_class, id_set);
+    get_parent().lookup_rec(
+      base_name_to_lookup, kind, identifier_class, id_set);
 }
 
 cpp_scopet::id_sett cpp_scopet::lookup_identifier(
-  const irep_idt &identifier,
-  cpp_idt::id_classt id_class)
+  const irep_idt &id,
```

Comment:
not sure why id is better than identifier

Reply:
It really just is different. I think the class member should not have been named "identifier" in the first place, but refactoring this when shadowing is still present seemed risky. I'd thus first get rid of shadowing (as is happening here) and then try to change names.